internal/commands/pull/command_test.go

```diff
@@ -317,6 +334,204 @@ Successfully pulled app down: app
 		assert.Nil(t, err)
 		assert.Equal(t, "<html><body>hello world!</body></html>", string(modifiedData))
 	})
+
+	t.Run("with a template ID to export with", func(t *testing.T) {
+
+		// Needs to have a successful export for templates
+		zipPkg, err := zip.OpenReader("testdata/test.zip")
+		assert.Nil(t, err)
+		defer zipPkg.Close()
+
+		var realmClient mock.RealmClient
+
+		realmClient.FindAppsFn = func(filter realm.AppFilter) ([]realm.App, error) {
+			return []realm.App{{ID: "appID", Name: "appName"}}, nil
+		}
+		realmClient.ExportFn = func(groupID, appID string, req realm.ExportRequest) (string, *zip.Reader, error) {
+			return "app_20210101", &zipPkg.Reader, nil
+		}
+
+		templateZipPkg1, err := zip.OpenReader("testdata/template_1.zip")
+		assert.Nil(t, err)
+		defer templateZipPkg1.Close()
+
+		templateZipPkg2, err := zip.OpenReader("testdata/template_2.zip")
+		assert.Nil(t, err)
+		defer templateZipPkg2.Close()
+
+		for _, tc := range []struct {
+			templateID       string
+			templateZip      *zip.Reader
+			expectedTemplate string
+		}{
+			{
+				templateID:  "template_1",
+				templateZip: &templateZipPkg1.Reader,
+				expectedTemplate: `{
+  "egg": "over easy",
+  "pancake": "blueberry",
+  "toast": "french"
+}`,
+			},
+			{
+				templateID:  "template_2",
+				templateZip: &templateZipPkg2.Reader,
+				expectedTemplate: `{
+  "egg": "scrambled",
+  "pancake": "plain",
+  "toast": "none"
+}`,
+			},
+		} {
+			t.Run(fmt.Sprintf("should export the corresponding template when the template ID %s is passed in", tc.templateID), func(t *testing.T) {
+				realmClient.CompatibleTemplatesFn = func(groupID, appID string) ([]realm.Template, bool, error) {
+					return []realm.Template{{ID: tc.templateID, Name: "some template"}}, true, nil
+				}
+
+				realmClient.ClientTemplateFn = func(groupID, appID, templateID string) (*zip.Reader, error) {
+					return tc.templateZip, nil
+				}
+
+				profile, teardown := mock.NewProfileFromTmpDir(t, "profile_name")
+				defer teardown()
+
+				_, ui := mock.NewUI()
+
+				cmd := &Command{inputs{Project: "some_project", LocalPath: "app", TemplateID: tc.templateID}}
+				assert.Nil(t, cmd.Handler(profile, ui, cli.Clients{Realm: realmClient}))
+
+				destination := filepath.Join(profile.WorkingDirectory, "app", local.FrontendPath)
+
+				_, err := os.Stat(destination)
+				assert.Nil(t, err)
+
+				testData, readErr := ioutil.ReadFile(filepath.Join(destination, fmt.Sprintf("%s.json", tc.templateID)))
+				assert.Nil(t, readErr)
+				assert.Equal(t, tc.expectedTemplate, string(testData))
+			})
+		}
+
+		t.Run("should export all selected compatible templates if no template ID is passed in and the app is made with a template", func(t *testing.T) {
```

Comment:
```suggestion
		t.Run("should export all selected compatible templates if no template id is passed in and the app is made with a template", func(t *testing.T) {
```

this should also probably call out that "auto confirm is set to true" somehow, b/c the behavior of this command would be different if not (and if we don't have test coverage for that scenario, seems like we'd want it)

Reply:
The only difference when AutoConfirm is true is there is no extra prompt for this, but I'll add it to the test description
`Would you like to export with a template?`